### ISSUE TITLE
Staging a harness configuration change (hook entry, settings file) as an insertable fragment risks a silent, total loss of a file's OTHER protections when the hand-merge goes wrong

### ISSUE THREAD

**Issue by Pfeifdanix** [rebelytics/one-skill-to-rule-them-all#173]
## Environment

- Claude Code, project-level `.claude/settings.json` carrying both a
  newly-staged `SessionStart` hook entry and pre-existing content: a
  command deny-list and a separate governance-protection hook.

## Symptom

Where a review or an in-session change proposes a harness configuration
change (a `SessionStart` or other hook entry, a settings-file addition —
as opposed to a skill file update, which this skill already stages as a
complete directory per "Delivering updated skills"), the natural
delivery shape is "here's the new key, add it under `hooks`" — a
fragment, staged and handed to the user with an instruction to insert it.

This is a materially different risk profile from staging a skill file.
A skill's `SKILL.md` is (almost always) the only thing that reads itself;
a corrupted staged copy fails to parse and nothing else is affected. A
harness's own configuration file is frequently NOT single-purpose: the
same `settings.json` that would receive the new hook entry also commonly
carries a command deny-list, other hooks, and permission rules — the
harness's actual protective layer. JSON has no partial-failure mode: a
single missing closing bracket, introduced during a manual "paste this
under `hooks`" merge, makes the WHOLE file fail to parse, which silently
disables every rule the file carries, not only the newly-added one.

Observed directly: a staged `SessionStart` hook was delivered as a bare
`hooks.SessionStart` key with the instruction "add this under `hooks`".
The hand-merge omitted the array's closing bracket. The resulting
`settings.json` was invalid JSON. Because the same file carried a
governance-protection hook and a command deny-list, BOTH were silently
disabled for as long as the file stayed invalid — a Bash command whose
path should have been blocked by the deny-list was allowed through while
the file was broken, and was correctly blocked again once the JSON was
fixed, confirming that the parse failure had disabled the protection
layer rather than merely failing to add the new hook. Separately, the
staged fragment was inserted into the wrong file (the project-level
settings rather than the user-level one the accompanying instructions
named as the target), and the hook's target script had not been copied
into the path the new hook entry pointed at — so even after the JSON was
fixed, the hook referenced a non-existent file.

## Proposed change

In `references/weekly-review.md`, "Delivering updated skills", and in
`references/environments.md`, immediately after "The same shape applies
to a blocked hook installation", add a rule distinguishing skill-file
staging from harness-configuration staging:

```markdown
**Harness configuration is a different delivery shape from a skill file,
and never ships as a fragment.** Where the change is not a skill update
but a change to the harness's own configuration (a hook entry, a
settings file), the target file typically also carries unrelated
protective rules (a deny list, a governance hook) that a hand-merge can
silently break. Stage the complete target file — produced from a fresh
read of the live file with the change already merged in — or a merge
command with a built-in validity check (`jq`, or a
`python3 -c "import json; json.load(open(path))"` check run immediately
after writing) — never "insert this fragment under the existing key".
State the install steps in a fixed order: copy any accompanying script
first, replace the settings file second, validate its JSON third —
including confirming the deny list and every other pre-existing rule
survived unchanged — and only then treat it as installed, with a test in
a fresh session as the last step. A parse error in a file that also
carries protective rules disables ALL of them silently, not only the
newly staged change.
```

## Why this matters generally

Any tool that stages a partial-file change for a human to hand-merge into
a shared configuration file is exposed to this, independent of which
harness or hook mechanism is involved: JSON (and most structured config
formats) fail closed on a syntax error, and a file that mixes a new
capability with pre-existing safety rules loses BOTH when the merge goes
wrong — silently, because nothing in the format itself signals which
rules were active before the edit.

**Comment by rebelytics** [comment]
Landed in v3.3.0. A harness config change is staged as the complete file, never as a fragment to hand-merge. Four reports from you in this release, all of them landed — thanks a lot.
